DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp

```diff
@@ -124,19 +161,32 @@ void ArrowCsvForeignStorage::read(const ChunkKey& chunk_key,
           varlen_offset += data[(sz / sizeof(uint32_t)) - 1];
         }
       } else {
-        std::memcpy(dest, bp->data(), sz = bp->size());
+        auto fixed_type = dynamic_cast<arrow::FixedWidthType*>(array_data->type.get());
+        if (fixed_type) {
+          std::memcpy(
+              dest,
+              bp->data() + (array_data->offset + offset) * (fixed_type->bit_width() / 8),
+              sz = size * (fixed_type->bit_width() / 8));
+        } else {
+          auto offsets_buffer =
+              reinterpret_cast<const uint32_t*>(array_data->buffers[1]->data());
+          auto string_buffer_offset = offsets_buffer[offset + array_data->offset];
+          auto string_buffer_size =
+              offsets_buffer[offset + array_data->offset + size] - string_buffer_offset;
+          std::memcpy(dest, bp->data() + string_buffer_offset, sz = string_buffer_size);
+        }
       }
     } else {
       // TODO: nullify?
       auto fixed_type = dynamic_cast<arrow::FixedWidthType*>(array_data->type.get());
       if (fixed_type) {
-        sz = array_data->length * fixed_type->bit_width() / 8;
-      } else
+        sz = size * (fixed_type->bit_width() / 8);
+      } else {
         CHECK(false);  // TODO: what's else???
```

Comment:
it's still here.. can we do something with this branch?

Reply:
It is unreachable as we don't know any cases, that can go that way. So it's ok to paste CHECK(false) in the unreachable branch.

Reply:
then, remove/reword the comment